### PR TITLE
CORTX-33288: Avoid redundant logs, improve log relevance.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
Signed-off-by: Shriya Deshmukh <shriya.deshmukh@seagate.com>

# Problem Statement
- Avoid redundant logs, improve log relevance.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by priyanka25081999 — APPROVED** [review]
LGTM

**Review by sachitanands — APPROVED** [review]
looks good

**Comment by shriya-deshmukh** [comment]
retest this please

File: src/rgw/setup/rgw.py [hunk 1]
```diff
@@ -434,7 +422,8 @@ def _create_rgw_user(conf: MappedConf):
         # Adding retry logic for user creation with given timeout value.
         retry_count = 0
         rc = -1
-        while(retry_count < const.USER_CREATION_MAX_RETRY_COUNT):
+        while (retry_count < const.USER_CREATION_MAX_RETRY_COUNT):
+            Log.info(f'Creating admin user, command : {create_usr_cmd}.')
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:426]: no lets not log admin user creation command, its little sensitive info.

File: src/rgw/setup/rgw.py [hunk 2]
```diff
@@ -273,7 +263,6 @@ def upgrade(conf: MappedConf, changeset_path: str):
         except Exception as e:
             raise SetupError(errno.EINVAL, f'Upgrade failed with error: {e}')
 
-        Log.info('Upgrade phase completed.')
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:276]: 'completed' phase logs are rather useful, instead we can think of removing phase 'started' logs, where ever exists. 
Phases which are not implemented yet, like reset, test, can be avoided with 'completed' logs though, which you have done above

File: src/rgw/setup/rgw.py [hunk 3]
```diff
@@ -985,19 +974,19 @@ def _update_svc_data_path_value(conf: MappedConf, client_section: str):
...
         # Create data path's default value e.g. /var/lib/ceph/radosgw/<cluster-id>
         data_path_default_value = const.SVC_DATA_PATH_DEFAULT_VALUE + cluster_id
         confstore_data_path_value = conf.get(const.SVC_DATA_PATH_CONFSTORE_KEY)
         if confstore_data_path_value is not None:
-           Log.info(f'Setting config key :{const.SVC_DATA_PATH_KEY} with value:{confstore_data_path_value}'
+           Log.debug(f'Setting KV pair {const.SVC_DATA_PATH_KEY}:{confstore_data_path_value}'
                f'at {client_section} section')
            Conf.set(Rgw._conf_idx, f'{client_section}>{const.SVC_DATA_PATH_KEY}', str(confstore_data_path_value))
         else:
-           Log.info(f'GConfig entry is missing for config key :{const.SVC_DATA_PATH_KEY}.'
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:998]: this would be useful  at INFO log level only, given default values being used.

**Review by cdeshmukh — CHANGES_REQUESTED** [review]
Overall good log optimizations i see in this PR

File: src/rgw/setup/rgw.py (hunk 2 above)
Comment (shriya-deshmukh) [reply src/rgw/setup/rgw.py:276]: We have added a log statement for **'{phase_name} phase completed'** in rgw_setup.py file , So here it is not required.

File: src/rgw/setup/rgw.py (hunk 3 above)
Comment (shriya-deshmukh) [reply src/rgw/setup/rgw.py:998]: Done

File: src/rgw/setup/rgw.py (hunk 1 above)
Comment (shriya-deshmukh) [reply src/rgw/setup/rgw.py]: Done

File: src/rgw/setup/rgw.py [hunk 4]
```diff
@@ -452,7 +441,7 @@ def _create_rgw_user(conf: MappedConf):
                     retry_count = retry_count + 1
                     continue
                 else:
-                    Log.error(f'"{create_usr_cmd}" failed with error {err}.')
+                    Log.error(f'RGW admin user creation failed with error {err}.')
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:444]: good you handled here too. Someone looking at the code can figure this out, but he/she will have to feed some values to it, which would take some efforts still. For debugging, we can do this feeding and figure out still

File: src/rgw/setup/rgw.py [hunk 5]
```diff
@@ -747,17 +733,17 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
...
         for machine_id in node_machine_ids:
             data_pod_hostnames.append(Rgw._get_cortx_conf(conf, const.NODE_HOSTNAME % machine_id))
 
-        Log.debug(f'collected all data pod hostnames from GConf : {data_pod_hostnames}')
+        Log.debug(f'Collected all data pod hostnames from GConf : {data_pod_hostnames}')
         return data_pod_hostnames
 
     @staticmethod
-    def _get_lock(consul_idx: str):
+    def _get_lock(conf: MappedConf, consul_idx: str):
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw.py:746]: why this change here?

File: src/rgw/setup/rgw.py (hunk 2 above)
Comment (shriya-deshmukh) [reply src/rgw/setup/rgw.py:276]: yes.

File: src/rgw/setup/rgw.py (hunk 5 above)
Comment (cdeshmukh) [reply src/rgw/setup/rgw.py:746]: ok

**Review by cdeshmukh — APPROVED** [review]
